src/pybind/mgr/prometheus/module.py

```diff
@@ -1378,17 +1378,17 @@ def respond() -> Optional[str]:
 
         self.metrics_thread.start()
 
+        cherrypy.config.update({
+            'server.socket_host': server_addr,
+            'server.socket_port': server_port,
+            'engine.autoreload.on': False
+        })
```

Comment:
This change super deceiving actually. If you want to allow binding to `::` or `0.0.0.0` you have to make this explicit. This change deceiving moves the `cherrypy.config.update` block before we're properly setting server_addr. 

With this change, `cherrypy` and `self.set_uri` would be called with two different `server_addr`. 

If you want to be able to bind to `::`, make it explicit please.